modules/msk/facets/main.nf

```diff
@@ -0,0 +1,80 @@
+process FACETS {
+    tag "$meta.id"
+    label 'process_medium'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'docker://mskcc/facets-suite:2.0.9':
+        'docker.io/mskcc/facets-suite:2.0.9' }"
+
+
+    input:
+
+    tuple val(meta), path(snp_pileup)  //  [ meta, ${prefix}.snp_pileup.gz]
+
+    output:
+    tuple val(meta), path("*_purity.seg")            , emit: purity_seg
+    tuple val(meta), path("*_purity.Rdata")          , emit: purity_rdata
+    tuple val(meta), path("*_purity.CNCF.png")       , emit: purity_png
+    tuple val(meta), path("*_purity.out")            , emit: purity_out
+    tuple val(meta), path("*_purity.cncf.txt")       , emit: purity_cncf_txt
+    tuple val(meta), path("*_hisens.seg")            , emit: hisens_seg
+    tuple val(meta), path("*_hisens.Rdata")          , emit: hisens_rdata
+    tuple val(meta), path("*_hisens.CNCF.png")       , emit: hisens_png
+    tuple val(meta), path("*_hisens.out")            , emit: hisens_out
+    tuple val(meta), path("*_hisens.cncf.txt")       , emit: hisens_cncf_txt
+    tuple val(meta), path("*.qc.txt")                , emit: qc_txt
+    tuple val(meta), path("*.gene_level.txt")        , emit: gene_level_txt
+    tuple val(meta), path("*.arm_level.txt")         , emit: arm_level_txt
+    tuple val(meta), path("*.txt")                   , emit: output_txt
+    path "versions.yml"                              , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    """
+    /usr/bin/facets-suite/run-facets-wrapper.R \
+        ${args} \
+        --sample-id ${prefix} \
+        --counts-file ${snp_pileup}
```

Comment:
I would like to keep the output file specific in case anyone needs to do downstream steps on a particular facets output file. I updated the logic and description so it works in normal and legacy mode.